src/jit/flowgraph.cpp

```diff
@@ -2415,25 +2415,51 @@ void Compiler::fgDfsInvPostOrderHelper(BasicBlock* block, BlockSet& visited, uns
     // Assume we haven't visited this node yet (callers ensure this).
     assert(!BlockSetOps::IsMember(this, visited, block->bbNum));
 
+    ArrayStack<DfsBlockEntry> stack(this);
+
+    DfsBlockEntry firstEntry(State_Pre, block);
```

Comment:
There's no need to introduce such locals, `stack.Push(DfsBlockEntry(State_Pre, block));` is enough.